src/core/manage.py

```diff
@@ -352,7 +363,27 @@ def upload_to(self, filename):
                         break
                     out_file.write(chunk)
         except Exception:
-            app.logger.critical("Upload failed!")
+            app.logger.exception("Upload failed!")
+            abort()
+
+    def download_update_file(self, filename, url):
+        import requests
+        import gzip
+
+        try:
+          app.logger.debug("start download of {0}".format(url))
+          r = requests.get(url, allow_redirects=True)
+          content_type = r.headers.get('Content-Type')
+          content_enc = r.headers.get('Content-Encoding')
+          if content_type == 'application/x-gzip' or content_enc == 'x-gzip':
+            content = gzip.decompress(r.content)
+          else:
+            if content_type != 'application/xml':
+              app.logger.debug("{0} does not point to an XML content-typed file".format(url))
+            content = r.content
+          open(filename, 'wb').write(content)
+        except Exception:
```

Comment:
you should also catch the Exception and print it to the user so it is easier to debug what failed (`except Exception as ex` - please use `ex` as the variable so it is consistent with other files). I suggest the use of traceback https://github.com/SK-CERT/Taranis-NG/blob/cca219beb864ed9bcdd6f6e637a230e3f864be04/src/collectors/collectors/web_collector.py#L518